### PR TITLE
304 magnolia migration

### LINKED ISSUE #304: Prepare DB Migration Script for Magnolia Model 2.1
Look at the finalized DSPEC add all keys for reference data to the migration script 

Image

### PR DESCRIPTION
**Pull request by lovelysandlonelys**
Migration Method to add and rollback reference location data and reference source data to the database.  I have... reused functions so let me know if you would rather we use a different approach. 

Please make sure to test both update and rollback methods when reviewing!
Update: 
image
NOTE: I made sure to check the database and anticipated rows were added

Rollback: 
image
NOTE: I made sure to check the database and anticipated rows were removed

Closes #304

### CODE REVIEW

**Review by matdenkas — APPROVED** [review]
WOOO!

File: tools/DatabaseMigration/2_1/init_data/magnoliaLocations.csv
Comment (bestrada33) [inline tools/DatabaseMigration/2_1/init_data/magnoliaLocations.csv:1]: Don't we also need the TCOON station info somewhere? Like the station IDs for the ingestion class. dataMapping I think it's called. For Port Lavaca and Port O'Connor.

File: tools/DatabaseMigration/2_1/2_1_DatabaseMigration.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+            for condition in conditions:
+                delete_stmt = delete(self.ref_dataLocation).where(
+                    *[getattr(self.ref_dataLocation.c, key) == value for key, value in condition.items()])
+                conn.execute(delete_stmt)
+                
+            conn.commit()
+
+        return results
+        
+    def remove_ref_dataSource(self, conditions: list[dict]) -> list[tuple]:
```
Comment (abeasley1722) [inline tools/DatabaseMigration/2_1/2_1_DatabaseMigration.py:144]: It's entirely up to you, but `remove_ref_dataSource` and `remove_ref_dataLocation` appear almost identical except for the specific variables `ref_dataLocation` or `ref_dataSource`. It's a bit difficult to explain, but you might find it simpler to merge both functions into one, naming it something like `remove_ref_data`, and then passing the necessary variable to that function.

Comment (lovelysandlonelys) [reply tools/DatabaseMigration/2_1/2_1_DatabaseMigration.py]: YOU'RE SO RIGHT FOR THIS

File: tools/DatabaseMigration/2_1/init_data/magnoliaLocations.csv
Comment (lovelysandlonelys) [reply tools/DatabaseMigration/2_1/init_data/magnoliaLocations.csv:1]: ahhh i think you are so right about this i completely forgot

**Review by bestrada33 — APPROVED** [review]
LGTM code wise. Didn't test

**Pull request merged**